src/ameba/rule/style/redundant_parentheses.cr

```diff
@@ -0,0 +1,65 @@
+module Ameba::Rule::Style
+  # A rule that disallows redundant parentheses around control expressions.
+  #
+  # For example, this is considered invalid:
+  #
+  # ```
+  # if (foo == 42)
+  #   do_something
+  # end
+  # ```
+  #
+  # And should be replaced by the following:
+  #
+  # ```
+  # if foo == 42
+  #   do_something
+  # end
+  # ```
+  #
+  # YAML configuration example:
+  #
+  # ```
+  # Style/RedundantParentheses:
+  #   Enabled: true
+  #   ExcludeTernary: false
+  #   ExcludeAssignments: false
+  # ```
+  class RedundantParentheses < Base
+    properties do
+      description "Disallows redundant parentheses around control expressions"
+
+      exclude_ternary false
+      exclude_assignments false
+    end
+
+    MSG = "Redundant parentheses"
+
+    # ameba:disable Metrics/CyclomaticComplexity
```

Comment:
can we reduce the complexity somehow and remove the directive? Maybe the case statement can be moved to a separate method?